internal/testrunner/runners/pipeline/benchmark.go

```diff
@@ -0,0 +1,388 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipeline
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/elasticsearch/ingest"
+	"github.com/elastic/elastic-package/internal/packages"
+	"github.com/elastic/elastic-package/internal/testrunner"
+)
+
+const (
+	// How many attempts to make while approximating
+	// benchmark duration by adjusting document count.
+	durationAdjustMaxTries = 3
+
+	// How close to the target duration for a benchmark
+	// to be is accepted.
+	durationToleranceSeconds = 0.5
+
+	// Same, but as a percentage of the target duration.
+	durationTolerancePercent = 0.9
+
+	// Minimum acceptable length for a benchmark result.
+	minDurationSeconds = 0.001 // 1ms
+
+	// How many top processors to return.
+	numTopProcs = 10
+)
+
+func BenchmarkPipeline(options testrunner.TestOptions) (*testrunner.BenchmarkResult, error) {
+	// Load all test documents
+	docs, err := loadAllTestDocs(options.TestFolder.Path)
```

Comment:
I guess that you need to prepare a spec PR first.

Reply:
will wait to open that one until we agree on the options and structure here